src/main/java/org/jenkinsci/plugins/github/webhook/RequirePostWithGHHookPayload.java

```diff
@@ -84,7 +85,11 @@ public void generateResponse(StaplerRequest req, StaplerResponse rsp, Object nod
                             throws IOException, ServletException {
                         RSAPublicKey key = new InstanceIdentity().getPublic();
                         rsp.setStatus(HttpServletResponse.SC_OK);
-                        rsp.setHeader(GitHubWebHook.X_INSTANCE_IDENTITY, new String(encodeBase64(key.getEncoded())));
+                        try {
+                            rsp.setHeader(GitHubWebHook.X_INSTANCE_IDENTITY, new String(encodeBase64(key.getEncoded()), "UTF-8"));
+                        } catch (UnsupportedEncodingException ex) {
+                            throw new IOException("Default UTF-8 encoding is not available", ex);
```

Comment:
this is not an IOException - it is a fatal JVM issue as UTF-8 is mandated by the JLS.